node/lib/Logger.ts

```diff
@@ -20,62 +22,89 @@ export class Logger implements ILogger {
             this.globalLogLevel = logLevel;
         }
 
-        let consoleTransport = new winston.transports.Console({
+        this.consoleTransport = new winston.transports.Console({
             colorize: true,
             level: this.globalLogLevel,
         });
 
         this.logger.configure({
             transports: [
-                consoleTransport,
+                this.consoleTransport,
             ],
         });
 
-        this.transportList.push(consoleTransport);
+        this.transportList.push(this.consoleTransport);
 
-        if (Logger.flag === true && filename ) {
-            let fileTransport = new winston.transports.File({
+        if (Logger.flag === true && filename) {
+            this.fileTransport = new winston.transports.File({
                 filename: filename,
                 handleExceptions: true,
                 level: this.globalLogLevel,
             });
 
             this.logger.configure({
                 transports: [
-                    consoleTransport,
-                    fileTransport,
+                    this.consoleTransport,
+                    this.fileTransport,
                 ],
             });
 
-            this.transportList.push(fileTransport);
+            this.transportList.push(this.fileTransport);
             Logger.flag = false;
         }
     }
 
-   public error(msg: string, logObject?: any): void {
-        this.logger.error(msg, logObject);
+   public error(msg: string, logObject?: any, loggingMetadata?: string): void {
+        this.logger.error(msg, logObject, loggingMetadata);
     }
 
-    public warn(msg: string, logObject?: any): void {
-        this.logger.warn(msg, logObject);
+    public warn(msg: string, logObject?: any, loggingMetadata?: string): void {
+       if(typeof(logObject === 'undefined') && typeof(loggingMetadata) !== 'undefined')
+       {
+           this.logger.error(msg, loggingMetadata);
```

Comment:
typo, was updating the error method 1st. Fixed

---
In reply to: [105340523](https://github.com/openT2T/opent2t/pull/38#discussion_r105340523) [](ancestors = 105340523)